st2common/st2common/models/api/keyvalue.py

```diff
@@ -44,6 +53,16 @@ class KeyValuePairAPI(BaseAPI):
                 'type': 'string',
                 'required': True
             },
+            'secret': {
```

Comment:
1. Yes
2. Yes for now. I am using encrypted as a display flag to intimate CLI or UI that the value sent out is encrypted. This is more for cosmetic purposes at this point. If I don't do that, st2 keyvalue get secret_key will display a value that is not immediately obvious that is encrypted. So `encrypted` field is more for user UX. This is the thing I wanted feedback on. 
3. Yes, that flag is how we know a key user wants to store needs to be encrypted. If you look at PUT APIs for secret vs non-secret, it might make sense.